guide.md

```diff
@@ -0,0 +1,388 @@
+pieces of CRUD. Each step in this guide will build on the previous, and is meant
+to be followed along. This guide assumes that you're using PostgreSQL on
+Rust nightly. We'll talk about how to use Diesel on stable Rust in the final
+chapter. Before we start, make sure you have PostgreSQL installed and running.
+
+The first thing we need to do is generate our project.
+
+```shell
+cargo new diesel_demo
+cd diesel_demo
+```
+
+We're going to use a [`.env`][dotenv-rust] file to manage our environment
+variables for us, and [`diesel_cli`][diesel-cli] to help set up the database.
+
+[dotenv-rust]: https://github.com/slapresta/rust-dotenv
+[diesel-cli]: https://github.com/sgrif/diesel/tree/master/diesel_cli
+
+```shell
+echo DATABASE_URL=postgres://localhost/diesel_demo > .env
+cargo install diesel_cli
+diesel setup
+```
+
+This will create our database for us (if it didn't already exist), and create an
+empty migrations directory that we can use to manage our schema. More on that
+later. Let's add Diesel to our `Cargo.toml`. We'll also add `dotenv` so we can
+use that same `.env` file for our environment variables.
+
+```toml
+[dependencies]
+diesel = "^0.5.0"
+diesel_codegen = { version = "^0.5.0", default-features = false, features = ["nightly", "postgres"] }
+dotenv = "^0.8.0"
+dotenv_macros = "^0.8.0"
+```
+
+We'll also add this to the top of `src/lib.rs`, which will allow us to use
+features from `diesel_codegen`.
+
+```rust
+#![feature(custom_derive, custom_attribute, plugin)]
+#![plugin(diesel_codegen, dotenv_macros)]
+```
+
+Now we're going to write a small CLI that lets us manage a blog (ignoring the
+fact that we can only access the database from this CLI...). The first thing
+we're going to need is a table to store our posts. Let's create a migration for
+that:
+
+```shell
+diesel migration generate create_posts_table
+```
+
+You should see output which looks something like this:
+
+```
+Creating migrations/20160202154039_create_posts/up.sql
+Creating migrations/20160202154039_create_posts/down.sql
+```
+
+Every migration consists of a way to run it, and a way to undo it. `down.sql`
+should always leave your database in the state it was in before running `up.sql`
+when possible. We'll edit our migration to contain the following:
+
+```sql
+-- up.sql
+CREATE TABLE posts (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR NOT NULL,
+  body TEXT NOT NULL,
+  published BOOLEAN NOT NULL DEFAULT 'f'
+)
+
+-- down.sql
+DROP TABLE posts
+```
+
+We can run our new migration with `diesel migration run`. It's usually a good
+idea to make sure that `down.sql` is correct. You can do a quick sanity check by
+doing `diesel migration redo`.
+
+OK enough SQL, let's write some Rust. We'll start by writing some code to show
+us the last 5 published posts. The first thing we need to do is establish a
+database connection.
+
+```rust
+// src/lib.rs
+#[macro_use]
+extern crate diesel;
+extern crate dotenv;
+
+use diesel::prelude::*;
+use diesel::pg::PgConnection;
```

Comment:
A few things here, most of which I assume were intentional and will be true in 0.5
* PgConnection lives in `diesel::connection::pg::PgConnection` right now
* "postgres" isn't currently a valid feature (listed above in the Cargo.toml)
* `diesel::Connection` has to be imported for the `Connection.establish` to be available

Again, I assume most of these things will be true by 0.5, just bringing them up